agent/agent.go

```diff
@@ -2547,6 +2550,15 @@ func (a *Agent) removeServiceLocked(serviceID structs.ServiceID, persist bool) e
 	a.logger.Printf("[DEBUG] agent: removed service %q", serviceID.String())
 
 	// If any Sidecar services exist for the removed service ID, remove them too.
+	err := a.removeServiceSidecars(serviceID, persist)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
`return a.removeServiceSidecars(serviceID, persist)` should have the same effect.

Reply:
Done